Source/GameMapEx.cpp

```diff
@@ -114,3 +114,29 @@ int GameMapEx::LoadMap(char *fileName)
 	
 	return func(p, 0, fileName);
 }
+
+void GameMapEx::CopyTileMap(int* tileMap, int xMin, int xMax, int yMin, int yMax)
+{
+	OP2Map *p = (OP2Map*)mapObj;
+	int **tileArray = (int**)(mapTileData);
+
+	int mapWidth = xMax - xMin;
+
+	for (int x = xMin; x < xMax; ++x)
+	{
+		for (int y = yMin; y < yMax; ++y)
+		{
+			int px = x - xMin;
+			int py = y - yMin;
+
+			int xLower = p->tileXMask & x;
+			int xUpper = xLower >> 5;
+			xLower &= 31;
+			xUpper <<= p->logTileHeight;
+			xUpper += y;
+			xUpper <<= 5;

```

Comment:
As per my earlier comment, this section of code could be extracted into an `inline` function. It converts the 2D `(x, y)` coordinates into a 1D array index. I believe OP2Utility has such a method.